sentry/src/lib.rs

```diff
@@ -90,8 +90,51 @@ pub use crate::init::{init, ClientInitGuard};
 
 /// Available Sentry Integrations.
 ///
-/// See the [`apply_defaults`] function for more information on which integrations are
-/// used by default.
+/// Integrations extend the functionality of the SDK for some common frameworks and
+/// libraries.  Integrations come two primary kinds: as event *sources* or as event
+/// *processors*.
+///
+/// Integrations which are *sources*, like e.g. the
+/// [`sentry::integrations::anyhow`](integrations::anyhow) integration, usually provide one
+/// or more functions to create new events.  They will usually provide their own extension
+/// trait exposing a new method on the [`Hub`].
+///
+/// Integrations which *process* events in some way usually implement the
+/// [`Itegration`](crate::Integration) trait and need to be installed when sentry is
+/// initialised.
+///
+/// # Installing Integrations
+///
+/// Processing integrations which implement [`Integration`](crate::Integration) need to be
+/// installed when sentry is initialised.  This is done using the
+/// [`ClientOptions::integrations`](crate::ClientOptions::integrations) field, which you can
```

Comment:
```suggestion
/// [`ClientOptions::add_integration`](crate::ClientOptions::add_integration) function, which you can
```

I would rather promote the function than to mess with the raw integrations vec, which should ideally be a private implementation detail, but that would break struct literals.

Reply:
Oh, I totally missed the function!  This is what we need more of these docs for :D  I'll update the docs with this.